pandas/core/internals/blocks.py

```diff
@@ -669,7 +669,9 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
             newb = self.copy() if copy else self
 
         if newb.is_numeric and self.is_numeric:
-            if newb.shape != self.shape:
+            # use values.shape, rather than newb.shape, as newb.shape
+            # may be incorrect for ExtensionBlocks.
+            if values.shape != self.shape:
```

Comment:
Yeah, I ran into this same issue when I was testing out a similar implementation.

I got around it by passing the argument `ndim=self.ndim` to the `make_block` function on line 664 above, which seems to get the job done and passes all relevant tests, though I didn't run the entire test suite.  Not familiar enough with this code though to say if that's a better (or even adequate) workaround.

For what it's worth, it looks like this `ndim` inference in `NonConsolidatableMixIn` is where things start to go wrong, at least for the length 1 `Series` case:
https://github.com/pandas-dev/pandas/blob/cf70d11a25ec59a5b630921d86759c7fa49884db/pandas/core/internals/blocks.py#L1780-L1785

Again, not familiar enough with this code to immediately see a fix, but maybe this is helpful to someone with more knowledge of this code than me?

Reply:
>  Not familiar enough with this code though to say if that's a better (or even adequate) workaround.

That seems better.